File: 1603/1603.java
/**
 * @Author: washing
 * @DateTime: 2021/3/19 10:29
 * @File: 1603.java
 * @Desc: 
 **/
class ParkingSystem {

    int[] space = new int[4];
    public ParkingSystem(int big, int medium, int small) {
        space[0] = 0;
        space[1] = big;
        space[2] = medium;
        space[3] = small;
    }
    
    public boolean addCar(int carType) {
        return --space[carType] >= 0;
    }
}

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem obj = new ParkingSystem(big, medium, small);
 * boolean param_1 = obj.addCar(carType);
 */

File: 1603/1603.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/3/19 10:29
# File: 1603.py
# Desc: 
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.dic = {1:big, 2:medium, 3:small}

    def addCar(self, carType: int) -> bool:
        if self.dic[carType] >= 1:
            self.dic[carType] -= 1
            return True
        else:
            return False


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)